src/x86/x86_generator.py

```diff
@@ -404,7 +404,7 @@ def run_on_test_case(self, test_case: TestCase) -> None:
                 for inst in bb:
                     if inst.has_mem_operand(True):
                         memory_instructions.append(inst)
-                    if inst.name in ["DIV", "REX DIV"]:
+                    if inst.name in ["DIV", "REX DIV", "IDIV", "REX IDIV"]:
```

Comment:
Have you tested it under default configuration, i.e. with exceptions disabled? I believe IDIV requires more different instrumentation than DIV